src/Desktop/UIAutomation/CustomObjects/Registrar.cs

```diff
@@ -72,17 +76,18 @@ private static UIAutomationType GetUnderlyingUIAType(CustomUIAPropertyType type)
             }
         }
 
-        internal static ITypeConverter CreateTypeConverter(CustomUIAPropertyType type)
+        internal static ITypeConverter CreateTypeConverter(CustomProperty prop)
         {
-            switch (type)
+            switch (prop.Type)
             {
                 case CustomUIAPropertyType.String: return new StringTypeConverter();
                 case CustomUIAPropertyType.Int: return new IntTypeConverter();
                 case CustomUIAPropertyType.Bool: return new BoolTypeConverter();
                 case CustomUIAPropertyType.Double: return new DoubleTypeConverter();
                 case CustomUIAPropertyType.Point: return new PointTypeConverter();
                 case CustomUIAPropertyType.Element: return new ElementTypeConverter();
-                default: throw new ArgumentException("Unset or unknown type", nameof(type));
+                case CustomUIAPropertyType.Enum: return new EnumTypeConverter(prop.Values);
+                default: throw new InvalidDataException("Unset or unknown type");
```

Comment:
I think I'd probably still stick with ArgumentException instead of InvalidDataException here. Is there value in including the numeric value of the unknown type in the error message?